apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerTransactionCreationOptionsTests.cs

```diff
@@ -150,6 +152,24 @@ public void WithIsDetached()
         Assert.False(options.IsDetached);
     }
 
+    [Fact]
+    public void WithReadLockMode()
+    {
+        var options = SpannerTransactionCreationOptions.ReadWrite.WithReadLockMode(TransactionReadLockMode.Optimistic);
+        Assert.Equal(TransactionReadLockMode.Optimistic, options.ReadLockMode);
+        options = options.WithReadLockMode(TransactionReadLockMode.Pessimistic);
+        Assert.Equal(TransactionReadLockMode.Pessimistic, options.ReadLockMode);
+    }
+
+    [Fact]
+    public void WithReadLockMode_ReadOnly()
+    {
+        var options = SpannerTransactionCreationOptions.ReadOnly.WithReadLockMode(TransactionReadLockMode.Unspecified);
```

Comment:
And when you start checking for partitioned DML, you also need to add a test for that here.